localMatchApp/sources/Login.cpp

```diff
@@ -0,0 +1,82 @@
+#include "Login.h"
+#include <iostream>
+
+using namespace std;
+
+Login::Login ( ) {
+  // ToDo: Connect to database.
+  // This console out is created for debug process
+  std::cout<<"object created"<<std::endl;
+}
+
+Login::~Login ( ) {
+  // ToDo: Disconnect to database.
+  // This console out is created for debug process
+  std::cout<<"object deleted"<<std::endl;
+}
+
+Login::ConnectionMessage Login::signUp ( const string& i_user, const string& i_password,
+                                         const string& i_repeatedPassword ) {
+  ConnectionMessage signUpConnectionState=ConnectionMessage::goodEntry;
+  // ToDo: Send and recive information to database.
+  // This string is just created for debug process
+  string userInDatabase = "Lol2018";
+
+  if ( !hasCorrectFormat ( i_password ) || !hasCorrectFormat ( i_repeatedPassword ) ) {
+    signUpConnectionState= ConnectionMessage::wrongPassword;
+  }
+
+  if ( i_password != i_repeatedPassword && (hasCorrectFormat ( i_password ) && hasCorrectFormat ( i_repeatedPassword ))) {
+    signUpConnectionState=ConnectionMessage::notEqualPasswords;
+  }
+
+  if ( i_user==userInDatabase || i_user.length()==0) {
```

Comment:
it is not necessary to check for the length of the username.
If it is zero, the hasCorrectFormat will return false.

Reply:
The "hasCorrectFormat" function just check the format of the password.